helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PerInstanceAccessor.java

```diff
@@ -542,4 +558,23 @@ public Response getHealthReportsOnInstance(
   private boolean validInstance(JsonNode node, String instanceName) {
     return instanceName.equals(node.get(Properties.id.name()).getValueAsText());
   }
+
+  private boolean validateDeltaInstanceConfigForUpdate(String clusterName, String instanceName,
+      ConfigAccessor configAccessor, InstanceConfig newInstanceConfig, boolean isDelete)
+      throws IllegalArgumentException {
+    InstanceConfig originalInstanceConfigCopy =
+        configAccessor.getInstanceConfig(clusterName, instanceName);
+    if (isDelete) {
+      for (Map.Entry<String, String> entry : newInstanceConfig.getRecord().getSimpleFields()
+          .entrySet()) {
+        originalInstanceConfigCopy.getRecord().getSimpleFields().remove(entry.getKey());
+      }
+    } else {
+      originalInstanceConfigCopy.getRecord().update(newInstanceConfig.getRecord());
+    }
+
+    return ConfigAccessor
```

Comment:
Is this sanity check only for topology? If yes, let's name it as topoogy check for the perInstanceAccessor, because the sanity check should also involve the "HELIX_PORT" something like that in the simple field.

Reply:
Updated.